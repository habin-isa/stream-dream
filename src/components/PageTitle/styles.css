body {
    margin: 0;
    padding: 0;
  }
  
  .container {
    display: flex;
    justify-content: center;
    align-content: center;
    flex-direction: column;
    height: 100vh;
    width: 100vw;
    background: #333;
  }
  
  h1 {
    text-align: center;
    text-transform: uppercase;
    font-family: Helvetica, Arial, sans-serif;
    font-size: 50px;
    letter-spacing: 1px; 
    color: #fcba03;
  }
  
  .animation {
    display: inline-block;
    animation: fadeIn 1.5s forwards;
    opacity: 0;
    transition-timing-function: cubic-bezier(0.785, 0.135, 0.15, 0.86);
  }
  
  .first {
    display: inline-block;
    animation: firstSpan 1.5s forwards cubic-bezier(0.785, 0.135, 0.15, 0.86);
    background: #333;
  }
  
  .layer {
    display: inline-flex;
    overflow: hidden;
  }
  
  .second {
    animation: secondSpan 1.5s forwards cubic-bezier(0.785, 0.135, 0.15, 0.86);
    font-weight: 100;
    z-index: -1;
    display: inline-block;
  }
  
  
  @keyframes fadeIn {
    0% {
      opacity: 0;
      transform: scale(1);
    } 
    50% {
      opacity: 1;
      transform: scale(1.1);
    }
    100% {
      transform: scale(1);
      opacity: 1;
    }
  }
  
  @keyframes firstSpan {
    0% {
      transform: translateX(50%);
    }
    60% {
      transform: translateX(50%);
    }
    100% {
      transform: translateX(0%);
    }
  }
  
  @keyframes secondSpan {
    0% {
      transform: translateX(-100%);
    }
    60% {
      transform: translateX(-100%);
    }
    100% {
      transform: translateX(-0%);
    }
  }