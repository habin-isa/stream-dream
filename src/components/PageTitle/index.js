import React from 'react';
import './styles.css'

const PageTitle = () => {
    return (
        <div className="container">
            <h1>
            <span className="animation">
                <span className="first">STREAM</span> 
                <span className="layer">
                <span className="second">DREAM</span>
                </span>
            </span>
            </h1>
        </div>
    )
};

export default PageTitle;