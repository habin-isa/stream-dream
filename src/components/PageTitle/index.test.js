import React from 'react';
import { render } from '@testing-library/react';
import PageTitle from '.';

describe('PageTitle', () => {
    it('should render', () => {
        const { container } = render(<PageTitle />);
        expect(container.firstChild).toMatchSnapshot();
    });
});
