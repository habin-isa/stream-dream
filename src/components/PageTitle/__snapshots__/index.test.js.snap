// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageTitle should render 1`] = `
<div
  class="container"
>
  <h1>
    <span
      class="animation"
    >
      <span
        class="first"
      >
        STREAM
      </span>
      <span
        class="layer"
      >
        <span
          class="second"
        >
          DREAM
        </span>
      </span>
    </span>
  </h1>
</div>
`;
